pkg/clients/request.go

```diff
@@ -69,9 +69,7 @@ func GetRequest(url string) ([]byte, error) {
 		return nil, err
 	}
 
-	if resp.StatusCode == http.StatusNotFound {
-		return nil, types.ErrNotFound{}
-	} else if (resp.StatusCode != http.StatusOK) && (resp.StatusCode != http.StatusAccepted) {
+	if (resp.StatusCode != http.StatusOK) && (resp.StatusCode != http.StatusAccepted) {
 		return nil, types.NewErrServiceClient(resp.StatusCode, bodyBytes)
 	}
```

Comment:
The refactored request.go looks correct. The intent of the ErrServiceClient is to allow a service client to return an error when calling its target service. The error type allows the http status code and any message returned from the failed call to be used by upstream logging (by conforming to the Error interface). Therefore no special dispensation as applies to http.StatusNotFound should be necessary.

Reply:
@jdharms your proposed fix can have side effects, because in some cases it's expected a `types.ErrNotFound`. Like here:
https://github.com/edgexfoundry/edgex-go/blob/master/internal/core/data/router.go#L339
or here:
https://github.com/edgexfoundry/edgex-go/blob/master/internal/core/data/valuedescriptor.go#L161

Reply:
@jduranf Thanks, I will look into addressing these side effects and will update when finished.

Reply:
@tsconn23 @jduranf I've updated my changes to (hopefully) fix all other places where this error value is relied upon.  I've tested with the blackbox tests and unit tests.